hypervisor/src/kvm/mod.rs

```diff
@@ -1349,6 +1408,315 @@ impl cpu::Vcpu for KvmVcpu {
         Ok(state.into())
     }
 
+    #[cfg(target_arch = "riscv64")]
+    ///
+    /// Returns the RISC-V vCPU core registers.
+    /// The `KVM_GET_REGS` ioctl is not available on RISC-V 64-bit,
+    /// `KVM_GET_ONE_REG` is used to get registers one by one.
+    ///
+    fn get_regs(&self) -> cpu::Result<StandardRegisters> {
+        let mut state = kvm_riscv_core::default();
+
+        let off = offset_of!(kvm_riscv_core, regs, user_regs_struct, pc);
+        let mut bytes = [0_u8; 8];
+        self.fd
+            .lock()
+            .unwrap()
+            .get_one_reg(riscv64_reg_id!(KVM_REG_RISCV_CORE, off), &mut bytes)
+            .map_err(|e| cpu::HypervisorCpuError::GetRiscvCoreRegister(e.into()))?;
+        state.regs.pc = u64::from_le_bytes(bytes);
+
+        let off = offset_of!(kvm_riscv_core, regs, user_regs_struct, ra);
```

Comment:
Nit: there is quite a lot of repetition here. You can use a macro to shorten the code. If you don't want to change this, it is fine, too.

Reply:
I have this in my todo list, my suggestion is we keep it this way here, since this happens for `aarch64` already. I was planing to introduce a `reg_calculator` (that kind of thing) in either `kvm-ioctls` or `vmm-sys-utils`, and come back to this :)

Reply:
Thank you for pointing this out, it's good to know others are thinking the way I was thinking.

I will take care of this :)